torchquad/integration/utils.py

```diff
@@ -177,7 +177,7 @@ def _check_integration_domain(integration_domain):
                     integration_domain,
                     " does not specify a valid integration bound.",
                 )
-            if bounds[0] > bounds[1]:
+            if anp.all(bounds[0] > bounds[1]):
```

Comment:
```suggestion
            if anp.any(bounds[0] > bounds[1]):
```
Shouldn't this be this way since we want to raise an exception if any domain dimension is wrong?

Reply:
> @ilan-gold So, to make sure I remember correctly: This allows integrating multiple integrands on different domains by transforming each integral into [0,1], right?

Kind of - it allows process the same integrand repeatedly over a grid (as per my previous work) but each part of the grid gets its own custom `[a,b]` domain.

>     * Does it work for the other integrators (Newton Cotes, MC)? Should it?

You have to write a custom integrator.  I did.

>     * Can we add a tiny example somewhere how this can be used? I don't know how to test it 🥲

The "tiny example" I have is 60 lines ATM.  We could probably bring that down to 50 if we used a non-gaussian integrator (I need about 10 lines for resizing the roots).  I think my issue with adding a small example is that it then becomes something that also needs to be "correct."  

>     * Should we maybe change the domain in one of the test cases to use this, otherwise it isn't really tested, is it?

We can do this to check for the exception, at least.  I think this would actually count for testing the thing - with a "different" domain, either it fails with the check (good) or it fails in some horrible way later without the check (also a good sign).  So I'm not sure we need a small example to test this.  But perhaps we could just have a small mock example that makes the check or doesn't, and then checks the resulting integration domain (or something like that) so we don't have to worry about correctness of the integrand evaluation etc.  i.e., isolating this.

Reply:
So to sum up, I think a "tiny example" could be a mock fixture that doesn't even do integration but just wraps the `calculate_grid` method.  We could also just test the `IntegrationGrid` directly using the existing fixture.  Whichever you prefer.